### DISCUSSION TITLE
photoprism database backup not working

### DISCUSSION THREAD

**Discussion by lopelisoli** [photoprism/photoprism#3381] [Q&A] [answered]
Hello, I'm currently running the _November 18, 2022_ version of PhotoPrism, and would like to upgrade to the newest release. In preparation for that, I want to make a backup of my MariaDB database with the following CLI command:
`docker-compose exec photoprism photoprism backup -i - > photoprism-db-20230506.sql`

When the command finishes, I observe the following:
1. The `photoprism-db-20230506.sql` file is nearly empty, containing only the following few lines (and no backup, per se):
```
[36mINFO[0m[2023-05-06T20:33:04Z] config: case-insensitive file system detected
[36mINFO[0m[2023-05-06T20:33:04Z] PhotoPrism® needs your support!  
[36mINFO[0m[2023-05-06T20:33:04Z] Visit https://photoprism.app/membership to learn more.
```
2. PhotoPrism outputs the following lines to the terminal where I have it running:
```
mariadb_1     | 2023-05-06 20:34:48 4 [Warning] Aborted connection 4 to db: 'photoprism' user: 'photoprism' host: '172.19.0.3' (Got an error reading communication packets)
mariadb_1     | 2023-05-06 20:34:48 5 [Warning] Aborted connection 5 to db: 'photoprism' user: 'photoprism' host: '172.19.0.3' (Got an error writing communication packets)
```

**What can I do to successfully create the database backup?**

**Comment by lastzero** [comment]
This could be one of the issues that have been fixed. Check the release notes.

**Comment by ippocratis** [comment, marked as answer]
Backup

docker-compose -f  compose-file.yml exec dbname mysqldump -uroot -pYOUR_MARIADB_ROOT_PASSWORD --all-databases > dump-$(date +%F_%H-%M-%S).sql

restore

docker-compose -f compose-file.yml exec -T dbname mysql -uroot -pYOUR_MARIADB_ROOT_PASSWORD  < mariadb-dump.sql

As per documentation other files like eg sidecar thumbnails etc need to be backuped also
https://docs.photoprism.app/getting-started/advanced/backups/

**Comment by lopelisoli** [comment]
@lastzero, I think my problem is indeed the one reported (and fixed) in issue [#3140](https://github.com/photoprism/photoprism/issues/3140).

And thank you @ippocratis - using the commands you suggested, I was able to back up my database, and then restore it after upgrading PhotoPrism.
